### DISCUSSION TITLE
Spatie\ElasticsearchQueryBuilder\Builder::search(): Return value must be of type array, Elastic\Elasticsearch\Response\Elasticsearch returned

### DISCUSSION THREAD

**Discussion by Parviz-Elite** [spatie/elasticsearch-query-builder#22] [Q&A]
elasticsearch/elasticsearch > v8.3.3
spatie/elasticsearch-query-builder > v2.0.0
PHP > v8.1.0
Laravel > v9.24.0

Class:
`private $client_endpoint;
    private $client_user;
    private $client_password;
    private $client_certificate;

    public ?Client  $client  = null;
    public ?Builder $builder = null;

    /**
     * Elastic constructor.
     */
    public function __construct () {
        $this->client_endpoint    = config('elastic.client_endpoint');
        $this->client_user        = config('elastic.client_user');
        $this->client_password    = config('elastic.client_password');
        $this->client_certificate = config('elastic.client_certificate');
        $this->initClient();
        $this->initBuilder();
    }

    /**
     * Init Elastic Driver
     *
     * @return void
     */
    private function initClient () : void {
        try {
            $this->client = ClientBuilder::create()
                                         ->setHosts([$this->client_endpoint])
                                         ->setBasicAuthentication($this->client_user, $this->client_password)
                                         ->setCABundle($this->client_certificate)
                                         ->build();
        } catch ( AuthenticationException $e ) {
            /*new Log('Error during connecting to Elastic Server (Authentication)', 'error', [
                'message' => $e->getMessage()
            ]);*/
        } catch ( Exception $e ) {
            /*new Log('Error during connecting to Elastic Server', 'error', [
                'message' => $e->getMessage()
            ]);*/
        }
    }

    /**
     * Init Elastic Query Builder
     *
     * @return void
     */
    private function initBuilder () : void {
        if ( empty($this->client) ) {
            return;
        }

        $this->builder = new Builder($this->client);
    }
`

Use:
`Route::get('test', static function () {
    $e = new App\Services\Elastic\Elastic();

    $res = $e->builder
        ->index('chat_data_conversations')
        ->addQuery(\Spatie\ElasticsearchQueryBuilder\Queries\MatchQuery::create('created_at', '2022-08-16 11:42:45'))
        ->search();

    dd($res);
});`

**Comment by hussonkevin** [comment]
The same for me, the spatie return type is not good :
**Spatie package :**
```
public function search(): array
{
   ....
  return $this->client->search($params);
}
```

**Elasticsearch package :**
```
/**
 * ...
 * @return Elasticsearch|Promise
 */
public function search(array $params = [])
```

**Reply by JurePotocnik** [reply]
Yes, this is the far as i went discovering the problem. Even my VS is saying there's a problem. Cannot pass through this,

**Comment by imdhemy** [comment]
@Parviz-Elite @hussonkevin @JurePotocnik 

Fixed in https://github.com/spatie/elasticsearch-query-builder/pull/24
